pkg/controller/storagecluster/storagecluster.go

```diff
@@ -405,6 +405,55 @@ func (c *Controller) validateCloudStorageLabelKey(cluster *corev1.StorageCluster
 	return nil
 }
 
+func (c *Controller) runPreflightCheck(cluster *corev1.StorageCluster) error {
+	check, ok := cluster.Annotations[pxutil.AnnotationPreflightCheck]
+	if !ok {
+		return nil
+	} else if check == "false" {
+		if cluster.Status.Phase == string(corev1.ClusterConditionTypePreflight)+string(corev1.ClusterOperationFailed) {
+			return fmt.Errorf("preflight check failed, please run it again")
+		}
```

Comment:
This should not result into returning an error, right? Shouldn't it just mean don't run the pre-flight check?

Reply:
idea is if it ran preflight and failed, in the next reconcile loop it should be still blocked. if no error returned here, it will proceed to the next step later. here user can fix their prerequisite, then trigger the preflight check again by changing the annotation, otherwise everything got blocked here.

Reply:
I think we'll need a doc for that, I could see myself getting tripped up by that: "I fixed what you said, why aren't you starting again??"